lib/schema/reportedState.js

```diff
@@ -0,0 +1,33 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2020, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+'use strict';
+
+var schema = require('./schema.js');
+
+var idFields = ['type', 'deviceId', 'time'];
+schema.registerIdFields('reportedState', idFields);
+
+var stateArraySchema = schema.isArrayWithValueSchema({
+  state: schema.isString
```

Comment:
In `platform`, there are a [few fixed states](https://github.com/tidepool-org/platform/blob/25bfa5aaf9702b1d6a0eebe08798285c33868183/data/types/state/reported/state.go#L14-L18) (ie, `state` must be one of the fixed states). If the state is anything other than one of the fixed states, then `state` should be "other", and an additional field called `stateOther` should exist, with a string of maximum length 100.
If you wanted to replicate this as simply as possible, you could do it like this:
```
var stateArraySchema = schema.isArrayWithValueSchema({
  state: schema.in('other'),
  stateOther: schema.isStringOfMaxLength(100)
});
```